### DISCUSSION TITLE
ECMA Script Support in the Various OS's

### DISCUSSION THREAD

**Discussion by nothingismagick** [tauri-apps/wry#202] [General]
Hi, this Discussion is so that we can track exactly what ECMASCRIPT is supported inside of the various Webviews. To add your data, please post a response to this discussion question that includes the following information (use this as a template:)

```
| | |
-- | -- 
OS & Version | xxx
HARDWARE | xxx
WRY Version | xxx
| | | 
ECMAScript 5 (2009) | ✔
ECMAScript 6 (2015) | ✔
ECMAScript 2016 | no
ECMAScript 2017 | no
ECMAScript 2018 | no
```

To get the results, please clone this repo, stay on the dev branch and then run 

`> cargo run --example detect_js_ecma`

- [ ] macOS 10.x.x
- [x] macOS 11.x.x
- [ ] windows 7
- [ ] windows 8
- [x] windows 10
- [x] linux (variety)

**Comment by nothingismagick** [comment]
| | |
-- | -- 
OS & Version | macOS 11.2.3
HARDWARE | m1 air
WRY Version | dev 0.8.0
| | | 
ECMAScript 5 (2009) | ✔
ECMAScript 6 (2015) | ✔
ECMAScript 2016 | ✔
ECMAScript 2017 | ✔
ECMAScript 2018 | ✔

**Comment by amrbashir** [comment]
| | |
-- | -- 
OS & Version | Windows 10 20H2
HARDWARE | x86_64 
WRY Version | dev 0.8.0
| | | 
ECMAScript 5 (2009) | ✔
ECMAScript 6 (2015) | ✔
ECMAScript 2016 | ✔
ECMAScript 2017 | ✔
ECMAScript 2018 | ✔

**Comment by FabianLars** [comment]
* Windows same as above


| | |
-- | --
OS & Version | Ubuntu 21.04 (libwebkit2gtk-4.0-dev=2.32.0)
HARDWARE | x86_64
WRY Version | dev 0.8.0
| | | 
ECMAScript 5 (2009) | ✔
ECMAScript 6 (2015) | ✔
ECMAScript 2016 | ✔
ECMAScript 2017 | ✔
ECMAScript 2018 | ✔

| | |
-- | -- 
OS & Version | Ubuntu 18.04 (libwebkit2gtk-4.0-dev=2.20.1)
HARDWARE | x86_64
WRY Version | dev 0.8.0
| | | 
ECMAScript 5 (2009) | ✔
ECMAScript 6 (2015) | ✔
ECMAScript 2016 | ✔
ECMAScript 2017 | ✔
ECMAScript 2018 | ✔

**Comment by Akryum** [comment]
Maybe it would be a good idea to come  up with a `browserslist` config here 🐈

**Reply by nothingismagick** [reply]
great idea

**Reply by nothingismagick** [reply]
Like this?

```
  "browserslist": [
    "es2015",
    "es2016",
    "es2017",
    "es2018"
  ],
```  

or is a tool like this more appropriate here:
https://github.com/wessberg/browserslist-generator#faq

**Reply by Akryum** [reply]
No that won't work. 😅 

---

```js
var { browsersWithSupportForEcmaVersion } = require("@wessberg/browserslist-generator")

const browserslist = browsersWithSupportForEcmaVersion("es2018")

console.log(JSON.stringify(browserslist, null, 2))
```

yields this `browserslist` config:

```json
[
  "and_chr >= 63",
  "chrome >= 63",
  "and_ff >= 58",
  "android >= 90",
  "edge >= 79",
  "samsung >= 8.2",
  "safari >= 11.1",
  "ios_saf >= 11.4",
  "opera >= 50",
  "firefox >= 58",
  "unreleased and_chr versions",
  "unreleased chrome versions",
  "unreleased and_ff versions",
  "unreleased android versions",
  "unreleased edge versions",
  "unreleased samsung versions",
  "unreleased safari versions",
  "unreleased ios_saf versions",
  "unreleased opera versions",
  "unreleased firefox versions"
]
```

**Reply by Akryum** [reply]
![image](https://user-images.githubusercontent.com/2798204/116000989-b3137e80-a5f2-11eb-8f9a-f345ea34d4c0.png)

**Comment by Pierstoval** [comment]
| | |
-- | -- 
OS & Version | Ubuntu 20.04 in Windows 10 pro via WSL 2
HARDWARE | x86_64
WRY Version | 0.8.0-dev (commit f0d0845 )
| | | 
ECMAScript 5 (2009) | ✔
ECMAScript 6 (2015) | ✔
ECMAScript 2016 | ✔
ECMAScript 2017 | ✔
ECMAScript 2018 | ✔

![image](https://user-images.githubusercontent.com/3369266/116001255-e0146100-a5f3-11eb-8c95-d3aa4de9bb52.png)
